### PR TITLE
enable github actions for PRs on all branches

### PR DESCRIPTION
**Pull request by CunliangGeng**
Major changes
- update github actions to run building and type check actions for PRs on **all branches**
- separate actions for building and sonarcloud
- delete unused github action documentation.yml

### CODE REVIEW

**Comment by CunliangGeng** [comment]
* **#229** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/229?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#228** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/228?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#227** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/227?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#226** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/226?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#224** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/224?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CunliangGeng and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .github/workflows/sonar-cloud.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          python3 -m pip install --upgrade pip setuptools wheel
+      - name: Install nplinker and its dependencies
+        run: |
+          python3 -m pip install .[dev]
+          install-nplinker-deps --run-on-github
+      - name: Check style against standards using ruff
+        run: ruff check .
+      - name: Run unit tests with coverage
+        run: pytest --cov --cov-report term --cov-report xml --junitxml=xunit-result.xml tests/unit
+      - name: Correct coverage paths
```
Comment (gcroci2) [inline .github/workflows/sonar-cloud.yml:57]: Why is this correction needed?

**Review by gcroci2 — APPROVED** [review]

File: .github/workflows/sonar-cloud.yml (hunk 1 above)
Comment (CunliangGeng) [reply .github/workflows/sonar-cloud.yml:57]: The `coverage.xml` has absolute paths to code files. This correction is to change absolute paths to relative paths by removing the `$PWD`.

**Comment by CunliangGeng** [comment]
### Merge activity

* **Mar 21, 9:38 AM EDT**: @CunliangGeng started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/227).
* **Mar 21, 9:39 AM EDT**: @CunliangGeng merged this pull request with [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/227).

**Pull request merged**